### ISSUE TITLE
feat: CodeBERT linear probe classifier pipeline (SSC v11 CLF-001..007)

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#243]
## Summary

Implement a linear probe classifier pipeline for CodeBERT encoder, as specified in SSC v11 Section 5.

**Depends on**: #242 (encoder support)

## Tasks

| ID | Task | Description | Time |
|----|------|-------------|------|
| CLF-001 | Embedding extraction | Forward pass over 17,942 corpus entries, cache 768-dim [CLS] embeddings as safetensors | 2 hrs |
| CLF-002 | Linear probe | Train Linear(768, 2) on cached embeddings (Level 0: seconds) | 1 hr |
| CLF-003 | Evaluation | MCC, accuracy, recall on test split (1,794 samples) with bootstrap CI | 1 hr |
| CLF-004 | Escalation | If MCC CI lower < 0.2, fine-tune top-2 layers (Level 1), then full (Level 2) | 4 hrs |
| CLF-005 | Baselines comparison | Compare against majority, keyword, linter baselines from bashrs | 1 hr |
| CLF-006 | Generalization test | 50 novel unsafe scripts from bashrs | 1 hr |
| CLF-007 | Confidence caching | Cache confidence scores for all 17,942 entries for conversation generation | 30 min |

## Escalation Ladder

| Level | Approach | Params Trained | Time |
|-------|----------|---------------|------|
| 0 | Linear probe (frozen CodeBERT) | 1,538 | Seconds |
| 1 | Fine-tune top-2 layers + head | ~15M | ~30 min |
| 2 | Full fine-tune all layers | 125M | ~1 hr |
| 3 | Continue-pretrain on shell + fine-tune | 125M | ~4 hrs |

## Ship Gate (C-CLF-001)

```
MCC_CI_lower > 0.2 AND accuracy > 0.935 AND generalization >= 50%
```

## Kill Criterion

If Level 3 fails C-CLF-001: STOP. Classifier adds no value over linter. Document honestly.

## Data

bashrs provides:
- `bashrs corpus export-splits --output ./data/` → train.jsonl, val.jsonl, test.jsonl
- `bashrs corpus baselines` → reference MCC/accuracy/recall
- `bashrs corpus generalization-tests` → 50 OOD scripts

## Provable Contract

`linear-probe-classifier-v1.yaml`:
- Frozen encoder weights during probe training
- Probability simplex (softmax sums to 1.0)
- Embedding determinism (bit-identical on repeated calls)

**Comment by noahgift** [comment]
## Stage 1 Infrastructure Complete

All CLF-001..007 code infrastructure implemented across 3 commits:

### Commits
- `931ecaa` feat: add EncoderModel, EncoderBlock, and LinearProbe (CLF-001..003, CLF-007)
- `93279bc` feat: add escalation ladder, baselines comparison, generalization test (CLF-004..006)

### What's implemented
| Task | Status | Description |
|------|--------|-------------|
| CLF-001 | ✅ | `EncoderModel` with `cls_embedding()`, `from_safetensors()` |
| CLF-002 | ✅ | `LinearProbe` with SGD training, Xavier init, class weights |
| CLF-003 | ✅ | MCC, accuracy, recall, precision, confusion matrix, `bootstrap_mcc_ci()` |
| CLF-004 | ✅ | `EscalationLevel` enum + `should_escalate()` decision logic |
| CLF-005 | ✅ | `compare_baselines()` against majority/keyword/linter |
| CLF-006 | ✅ | `generalization_test()` on novel OOD embeddings |
| CLF-007 | ✅ | `compute_confidence_scores()` for all samples |

### Test Coverage
31 tests across 3 modules, all passing:
- `encoder_block::tests` — 4 tests (shape, finite, layer_idx, parameter count)
- `encoder::tests` — 5 tests (forward shape/finite, CLS shape/determinism, param count)
- `linear_probe::tests` — 22 tests (probe, MCC, baselines, escalation, generalization, ship gate)

### Remaining Work
Actual model training requires:
1. Download CodeBERT weights (`microsoft/codebert-base`)
2. Export bashrs corpus splits (`bashrs corpus export-splits`)
3. Run embedding extraction + linear probe training
4. Evaluate and escalate if needed

The infrastructure is ready — the remaining work is operational (model download + training run).

**Comment by noahgift** [comment]
## PV-003 Complete: Contract Falsification Tests Bound

Added 12 falsification tests + 3 proptests (`tests_ssc_contract_falsify.rs`) that bind to the 4 SSC YAML contracts:

### Contract Coverage

| Contract | Tests | Status |
|----------|-------|--------|
| `bidirectional-attention-v1.yaml` | FALSIFY-BIATT-001..003 | ✅ All pass |
| `linear-probe-classifier-v1.yaml` | FALSIFY-PROBE-001..003 | ✅ All pass |
| `encoder-forward-v1.yaml` | FALSIFY-ENC-001..002 | ✅ All pass |
| `learned-position-embedding-v1.yaml` | FALSIFY-POS-001 | ✅ All pass |

### Proptests
- `falsify_probe_002_prop`: Random embeddings → valid probabilities (100+ cases)
- `falsify_probe_003_prop`: Random H×K → correct param count (100+ cases)
- `falsify_enc_001_prop`: Random seq_len → correct output shape (100+ cases)

### Audit
```
pv audit bidirectional-attention-v1.yaml → 0 findings
pv audit linear-probe-classifier-v1.yaml → 0 findings
pv audit encoder-forward-v1.yaml → 0 findings
pv audit learned-position-embedding-v1.yaml → 0 findings
```

Commit: 5bbe427
